apps/desktop/desktop/src/main/java/bisq/desktop/main/content/components/BondedRoleBadge.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.desktop.main.content.components;
+
+import bisq.bonded_roles.BondedRoleType;
+import bisq.desktop.common.utils.ImageUtil;
+import bisq.desktop.components.controls.BisqTooltip;
+import bisq.i18n.Res;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+import java.util.List;
+
+public class BondedRoleBadge extends HBox {
+    private final BisqTooltip tooltip = new BisqTooltip();
+
+    public BondedRoleBadge(boolean useLargeIcon) {
+        ImageView badgeIcon;
+        if (useLargeIcon) {
+            badgeIcon = ImageUtil.getImageViewById("moderator-badge-large");
+            setAlignment(Pos.BOTTOM_LEFT);
+            setPadding(new Insets(0, -5, 5, 0));
+        } else {
+            badgeIcon = ImageUtil.getImageViewById("moderator-badge");
+            setPadding(new Insets(0.5, -6, 0, 0));
+        }
+        getChildren().add(badgeIcon);
+
+        setVisible(false);
+        setManaged(false);
+
+        Tooltip.install(this, tooltip);
+    }
+
+    public void setUserProfileBondedRoles(List<BondedRoleType> userBondedRoles) {
```

Comment:
Better use as Set as we don't require order but expect no duplication.

We don't have a strict guideline for such naming conventions but I think its better to call it applyBondedRoleTypes instead of setUserProfileBondedRoles as it is not a pure setter (also the name does not refer to the value - BondedRoleTypes). With apply prefix it gives a hint that more happens as setting a property (which is actually not happening here).